### DISCUSSION TITLE
How to convert datetime64[ns] effective to other timezones?

### DISCUSSION THREAD

**Discussion by Rumpen** [vaexio/vaex#1675] [Q&A] [answered]
Hi,

we have a basic Dataframe like this:

![image](https://user-images.githubusercontent.com/35268568/139924662-cfbce86e-68c3-4894-bb86-f717506ecd59.png)

The process time is in UTC and we want to group by hour and aggregate by total_net using mean:
 
`df['hour'] = df.process_time.dt.hour`
`group_by = df.groupby(by=df['hour'], sort="hour asc", agg={'aggregation': vaex.agg.mean(df['total_net'])})`

Result would be:

![image](https://user-images.githubusercontent.com/35268568/139925172-c0281a43-072d-4d2a-ba97-3d01e7e5f0fb.png)

![image](https://user-images.githubusercontent.com/35268568/139927597-e1af76c1-1de2-4139-94c0-af04a7a3e237.png)

Are there any best practices for converting the UTC timestamps to other Timezones using native vaex (like pandas .tz_localize and .tz_convert), e.g. User wants to see the aggregation in Timezone "Europe/Berlin" instead of "UTC"?

**Comment by JovanVeljanoski** [comment, marked as answer]
No native way i am afraid. I guess your best bet is to either do that prior to loading the data with vaex, or use some function (perhaps that of pandas) with `apply`. I vaex `apply` works in parallel, but best to be avoided if not necessary. 

If this is a one-time thing conversion, you can  "pay the price" with apply, and then maybe `df.materialize` the new column or store it to disk.

**Reply by Rumpen** [reply]
Thanks for the quick response and the good work on vaex!
